packages/reports/addon/components/filter-values/dimension-select.ts

```diff
@@ -110,16 +115,33 @@ export default class DimensionSelectComponent extends Component<DimensionSelectC
    */
   @task({ restartable: true })
   *searchDimensionValues(term: string): TaskGenerator<NaviDimensionModel[] | undefined> {
+    const { dimensionColumn } = this;
     const searchTerm = term.trim();
     this.searchTerm = searchTerm;
     if (!searchTerm) {
       return undefined;
     }
+
     yield timeout(this.isSmallCardinality ? SEARCH_DEBOUNCE_OFFLINE_MS : SEARCH_DEBOUNCE_MS);
     const dimensionResponse: NaviDimensionResponse = yield taskFor(this.naviDimension.search).perform(
       this.dimensionColumn,
       searchTerm
     );
-    return dimensionResponse.values;
+    if (dimensionResponse.values.map((each) => each.value).includes(searchTerm)) {
+      return dimensionResponse.values;
+    }
+
+    const dimension: ManualInput[] = [];
+    const dimensionModelFactory = getOwner(this).factoryFor('model:navi-dimension');
+    const value = searchTerm.toLowerCase() as string | number;
+    dimension.push(
+      dimensionModelFactory.create({
+        value,
+        dimensionColumn,
+        manualInputEntry: true,
+        manualQuery: searchTerm.toLowerCase(),
+      })
+    );
+    return [dimension[0], ...dimensionResponse.values];
```

Comment:
Do we need the array? Can we do `[manualInputEntry, ...dimensionResponse.values];`